bin/install-theos

```diff
@@ -0,0 +1,598 @@
+	else
+		update "\$THEOS has not been set. Setting now..."
+
+		if [[ "$SHELL_ENV" == "unknown" ]]; then
+			error "Current shell ($CSHELL) is unsupported by this installer. Please set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 7
+		fi
+
+		# Set $THEOS
+		echo "export THEOS=~/theos" >> "$SHELL_ENV"
+		source "$SHELL_ENV"
+		if [[ "$THEOS" ]]; then
+			update "\$THEOS has been successfully set to '$THEOS'!"
+		else
+			error "\$THEOS is still not set despite our efforts. Try restarting your shell and/or running the installer again. If that doesn't do the trick, you may have to set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 8
+		fi
+	fi
+
+	# Get Theos
+	update "Checking for Theos install..."
+	if [[ -d "$THEOS" && "$(ls -A "$THEOS")" ]]; then
+		update "Theos appears to already be installed. Checking for updates..."
+		$THEOS/bin/update-theos
+	else
+		update "Theos does not appear to be installed. Cloning now..."
+		git clone --recursive https://github.com/theos/theos.git $THEOS && update "Git clone of Theos was successful!" || (error "Theos git clone command seems to have encountered an error. Please see the log above." && exit 9)
+	fi
+
+	# Get patched sdks
+	update "Checking for patched sdks..."
+	if [[ -d "$THEOS/sdks/" && "$(ls -A "$THEOS/sdks/" | grep sdk)" ]]; then
+		update "SDKs appear to already be installed."
+	else
+		update "SDKS do not appear to be installed. Installing now..."
+		curl -LO https://github.com/theos/sdks/archive/master.zip
+		TMP=$(mktemp -d)
+		unzip master.zip -d $TMP
+		mv $TMP/sdks-master/*.sdk $THEOS/sdks
+		rm -r master.zip $TMP
+
+		if [[ -d "$THEOS/sdks/" && "$(ls -A "$THEOS/sdks/" | grep sdk)" ]]; then
+			update "SDKs successfully installed!"
+		else
+			error "Something appears to have gone wrong. Please try again."
+			exit 10
+		fi
+	fi
+}
+
+darwin-mobile() {
+	common "Platform: Darwin (Mobile)"
+
+	# Check for sudo (not installed by default on some jbs)
+	if ! command -v sudo &> /dev/null; then
+		error "Please install 'sudo' in your package manager before proceeding with the installation."
+		exit 3
+	fi
+
+	# Determine iOS version
+	OLD=0
+	KERNEL_VER=$(sysctl kern.osrelease | sed 's/[^0-9]*//g')
+	if [[ $KERNEL_VER < 1800 ]]; then
+		OLD=1
+	fi
+
+	# Dependencies
+	update "Preparing to install dependencies. Please enter your password if prompted:"
+	if [[ $OLD == 1 ]]; then
+		read -p "Do you have 'https://coolstar.org/publicrepo/', 'https://repo.bingner.com/', and 'http://apt.thebigboss.org/repofiles/cydia/' installed in the sources of your jailbreak's primary package manager? [y/n]" ready
+		if [[ "${ready,,}" == "y" || "${ready,,}" == "yes" ]]; then
+			sudo apt install org.theos.dependencies -y --allow-unauthenticated --allow-downgrades && update "Dependencies have been successfully installed!" || (error "Dependency install command seems to have encountered an error. Please see the log above." && exit 5)
```

Comment:
`apt-get` would be a safer bet for legacy, I think we didn’t get `apt` on Telesphoreo.

Reply:
Switched 'legacy' dep install to apt-get and kept the others as apt.